content/blog/centralized-policy-violations/index.md

```diff
@@ -0,0 +1,67 @@
+---
+title: "Introducing the Centralized Policy Violations Page: Streamlined Compliance Management"
+allow_long_title: true
+date: 2024-08-26T09:00:00-07:00
+draft: false
+meta_desc: The new Centralized Policy Violations page in Pulumi offers enhanced visibility and control by consolidating all policy violations.
+meta_image: meta.png
+authors:
+    - meagan-cojocar
+tags:
+    - releases
+    - policy
+---
+
+We’re excited to introduce a new page in Pulumi Cloud designed to simplify policy management and compliance: the centralized Policy Violations page. This powerful addition provides a single, comprehensive view of all policy violations across your organization, helping you maintain control and visibility over your infrastructure.
+
+Pulumi’s Policy as Code (PaC) feature, [CrossGuard](/docs/using-pulumi/crossguard), allows you to define and enforce policies for your infrastructure using familiar programming languages. By integrating these policies directly into your infrastructure code, you can automate compliance checks, gate deployments to ensure security best practices, and maintain governance across all your providers.
+
+**Note:** Pulumi CrossGuard is available in the Pulumi Business Critical edition.
+
+## Centralized Policy Violations: Why It Matters
+
+In large and complex environments, ensuring compliance across multiple projects and stacks can be challenging. The new Policy Violations page addresses this challenge by consolidating all violations into one place, allowing you to quickly assess the overall compliance of your infrastructure and separate signal from noise. This streamlined view not only saves time but also reduces the risk of missing critical issues.
+
+By having all violations on a single page, organization admins can efficiently prioritize remediation efforts and identify patterns in non-compliance. They can triage the violations and send the critical ones to teams to resolve.
+
+{{< video title="Centralized Policy Violations Overview" src="policy-violations-overview.mp4" autoplay="false" loop="false" >}}
+
+## Grouping Violations: Enhanced Insights Across Multiple Dimensions
+
+One of the standout features of the new page is the ability to group violations by various fields, including Policy Pack, Policy, Project, Stack, Resource, Reason, Enforcement Level, Version, and Violation Date. This flexibility allows you to gain deeper insights into your infrastructure’s compliance landscape.
```

Comment:
You currently can not group by reason, violation date, version, or update. I can change this if youd like for all of them but the violation date might look odd as the grouping would be down to the exact timestamp (we could update this to be by the string groupings but it might take some time to get right).

Reply:
Any reason why we prevent grouping by those?

Reply:
The cardinality on the reason one will be very high and be a very expensive query when we go to the backend because of how long the text is (it has to text match on all of those rows to build the results). The version and update was just from my initial thoughts so I can change those pretty easily. The violation date would need some more work to get right as the actual data is down to the timestamp where as we show a "rounded" string in the grid. The grouping would be based on that timestamp value so we would need to customize it to group by the strings instead.

Reply:
Reason maps with Policy 1:1 I believe, so I don't know that it should be treated separately. If its possible, I think I would make everything groupable except Violation Date, similar to the filters decision we made. This isn't urgent, but if we are going to do it I might leave in the blog and we do it post-launch.

Reply:
I agree with Meagan. Reason actually has very low cardinality.  The main challenge for grouping by reason is just that i t is ugly as a group.  Ideally if they grouped by Reason we would group by Policy instead and explain why, but that might be an awkward UX.

Reply:
Ok I can change them to allow grouping on all fields but Violation Date. I have some concerns about backend performance down the road but the data if the data size grows a lot, it is small right now.